src/notebook/notebookFeature.ts

```diff
@@ -202,19 +202,17 @@ export class JuliaNotebookFeature {
 
     private updateNotebookWithSelectedKernel(notebook: vscode.NotebookDocument, version: semver.SemVer) {
         // Dont edit in place, create a copy of the metadata.
-        const nbmetadata: JupyterNotebookMetadata = JSON.parse(JSON.stringify((notebook.metadata || { custom: { metadata: {} } })));
-        nbmetadata.custom.metadata.kernelspec = {
+        const kernelspec = {
             display_name: `Julia ${version}`,
             language: 'julia',
             name: `julia-${version.major}.${version.minor}`
         }
-        nbmetadata.custom.metadata.language_info = {
-            name: 'julia',
-            version: `${version}`,
-            mimetype: 'application/julia',
-            file_extension: '.jl'
+        if (notebook.metadata?.custom?.metadata?.language_info?.name) {
+            notebook.metadata.custom.metadata.language_info.name = 'julia'
+        }
+        if (this.vscodeIpynbApi) {
+            this.vscodeIpynbApi.setKernelSpec(notebook.uri, kernelspec)
```

Comment:
@davidanthoff Yes you are right, this should have been `setMetadata` or `setNotebookMetadata`.
Apologies for the oversight.

Reply:
@DonJayamanne is there any workaround that we could use, or do we have to wait for VS Code 1.61 for this to work?

Reply:
We wanted to create a limited api targeted just for this use case. My impression from the discussion in microsoft/vscode#131219 was setting the `kernelSpec` was all that was required to unblock you. If we added the ability to set language info too, would that fully address your use case then?

Do you want to try updating the api that VS Code exposes to take a `languageInfo` parameter and submit a PR if it looks good. The addition should be fairly straightforward. If you confirm that fixes the problem, we could ship the PR as part of a 1.60.1 build within a week or so

Reply:
> setting the kernelSpec was all that was required to unblock you

Well, anything that allows us to produce valid notebook files :) I should have looked more carefully at the PR to spot that it was missing the language info stuff.

> If we added the ability to set language info too, would that fully address your use case then?

Yes, I'm opening a PR now.

Reply:
@mjbvz, see https://github.com/microsoft/vscode/pull/132298.